course-v2/assets/js/mobile_course_drawers.ts

```diff
@@ -0,0 +1,35 @@
+export const initCourseDrawersClosingViaSwiping = () => {
+  elementSwiping("mobile-course-nav", "mobile-course-nav-toggle", "L")
+  elementSwiping("course-info-drawer", "mobile-course-info-toggle", "R")
+}
+
+/**
+ * It adds touch event listener to the element and clicks the button when swiped in the mentioned direction.
+ *
+ * @param {string} elementId element on which touch eventlistenter is added.
+ * @param {string} buttonId This button will be clicked on swiping
+ * @param {string} swipeDirection L= Swipe left, R= Swipe Right
+ */
+const elementSwiping = (
+  elementId: string,
+  buttonId: string,
+  swipeDirection: string
+) => {
+  const element = document.getElementById(elementId)
+  let touchstartX = 0
+  let touchendX = 0
+
+  element?.addEventListener("touchstart", e => {
+    touchstartX = e.changedTouches[0].screenX
+  })
+
+  element?.addEventListener("touchend", e => {
+    touchendX = e.changedTouches[0].screenX
+    if (swipeDirection === "R") {
```

Comment:
Overall looks good, this is just a nitpick and you can choose to implement this. Since you are already passing a swipeDirection parameter into the function, we can just pass the "<" and ">" as parameters into the function and evaluate the valid direction.
Generally eval function is considered `evil` but since we are generating strings ourselves, we would not face any issues.
[Reference for eval function](https://stackoverflow.com/questions/197769/when-is-javascripts-eval-not-evil) 
```suggestion
   if (eval(`${touchendX} ${swipeDirection} ${touchstartX}`)) document.getElementById(buttonId)?.click()
    
```

Reply:
Yeah, I had this in mind but did not use it deliberately:
1. since it's considered quite bad ("evil") due to potential security concerns (though our string is hardcoded)  
2. to have better readability of the code. 

But  I agree with you that it will make the code simpler. 
@ChristopherChudzicki Do you have any thoughts on this?

Reply:
@MAbdurrehman12, You also dont need to nest if blocks since you can just use the `&&` operator with the child if block.

Reply:
@MAbdurrehman12 We can also dont need to use a nested if since we can just use the "&&" with the child if and make it single level block. This would make the code a bit for readable.